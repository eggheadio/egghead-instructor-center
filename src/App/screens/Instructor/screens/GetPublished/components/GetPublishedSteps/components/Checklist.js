import React from 'react'
import MoreInfo from '../../MoreInfo'

const Checklist = ({items}) => (
  <div>
    {items.map((item, index) => (
      <div
        key={index}
        className='flex items-center b mb3'
      >
        <div className='mr2'>
          {item.isComplete
            ? <span className='fa fa-check-square-o fa-lg gray' />
            : <span className='fa fa-square-o fa-lg green' />
          }
        </div>
        <div>
          <div>
            <span className={item.isComplete
              ? 'strike gray'
              : 'green'
            }>
              {item.description}
            </span>
            {item.moreInfoUrl
              ? <span className='ml2'>
                  <MoreInfo url={item.moreInfoUrl} />
                </span>
              : null
            }
          </div>
        </div>
      </div>
    ))}
  </div>
)

Checklist.propTypes = {
  items: React.PropTypes.arrayOf(React.PropTypes.shape({
    isComplete: React.PropTypes.bool.isRequired,
    description: React.PropTypes.string.isRequired,
    moreInfoUrl: React.PropTypes.string,
  })).isRequired,
}

export default Checklist
