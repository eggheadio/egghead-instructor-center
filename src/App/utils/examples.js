export default {
  text: {
    short: 'Lorem ipsum',
    medium: 'Lorem ipsum dolor sit amet, urna erat feugiat egestas, metus molestie, adipiscing sed nisl et, dictumst erat sem erat.',
  },
  url: 'https://egghead.io',
  markdown: 'Lorem ipsum *dolor*',
}
