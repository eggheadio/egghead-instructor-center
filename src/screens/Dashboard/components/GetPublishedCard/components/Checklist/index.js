import React from 'react'
import {Link} from 'react-router-dom'
import {map} from 'lodash'
import {Icon, List} from 'egghead-ui'
import MoreInfo from './components/MoreInfo'

export default ({items}) => (
  <List items={map(items, (item, index) => (
    <div 
      key={index}
      className='flex items-center justify-between'
    >

      <div className='flex items-center'>

        <div className='mr3'>
          {index + 1}
        </div>

        <span className={item.isComplete
          ? `strike gray-secondary`
          : 'base'
        }>
          {item.description}
        </span>

      </div>

      {item.isComplete
        ? <Icon 
            type='check' 
            color='green'
          />
        : <div>
            {item.moreInfoUrl
              ? <span className='ml2'>
                  <MoreInfo url={item.moreInfoUrl} />
                </span>
              : null
            }
            {item.action
              ? <Link 
                  to={item.action}
                  className='ml2'
                >
                  <Icon 
                    type='chevron-right' 
                    size='4'
                  />
                </Link>
              : null
            }
          </div>
      }

    </div>
  ))} />
)
