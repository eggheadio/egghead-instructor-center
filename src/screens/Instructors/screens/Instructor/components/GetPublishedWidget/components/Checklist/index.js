import React, {PropTypes} from 'react'
import {Link} from 'react-router-dom'
import {map} from 'lodash'
import {Icon} from 'egghead-ui'
import List from 'components/List'
import MoreInfo from './components/MoreInfo'

const Checklist = ({items}) => (
  <List items={map(items, (item, index) => (
    <div 
      key={index}
      className='flex items-center b'
    >
      <div className='mr2'>
        {item.isComplete
          ? <Icon
              type='box-check'
              color='light-gray'
            />
          : <Icon
              type='box'
              color='gray'
            />
        }
      </div>
      <div>
        <div>
          <span className={item.isComplete
            ? 'strike light-gray'
            : 'dark-gray'
          }>
            {item.description}
          </span>
          {item.moreInfoUrl
            ? <span className='ml2'>
                <MoreInfo url={item.moreInfoUrl} />
              </span>
            : null
          }
          {item.action
            ? <Link 
                to={item.action}
                className='ml2'
              >
                <Icon type='arrow-right' />
              </Link>
            : null
          }
        </div>
      </div>
    </div>
  ))} />
)

Checklist.propTypes = {
  items: PropTypes.arrayOf(PropTypes.shape({
    isComplete: PropTypes.bool.isRequired,
    description: PropTypes.string.isRequired,
    actionUrl: PropTypes.string,
    moreInfoUrl: PropTypes.string,
  })).isRequired,
}

export default Checklist
