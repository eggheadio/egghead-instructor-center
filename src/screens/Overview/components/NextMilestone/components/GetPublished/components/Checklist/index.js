import React, {PropTypes} from 'react'
import {Link} from 'react-router-dom'
import {map} from 'lodash'
import Icon from 'components/Icon'
import MoreInfo from './components/MoreInfo'

const Checklist = ({items, instructorId}) => (
  <div>
    {map(items, (item, index) => (
      <div
        key={index}
        className='flex items-center b mb3'
      >

        <div className='mr2'>
          {item.isComplete
            ? <Icon
                type='step-complete'
                size='2'
                className='gray'
              />
            : <Icon
                type='step-incomplete'
                size='2'
                className='green'
              />
          }
        </div>

        <div>
          <div>
            <span className={item.isComplete
              ? 'strike gray'
              : 'green'
            }>
              {item.description}
            </span>
            {item.moreInfoUrl
              ? <span className='ml2'>
                  <MoreInfo url={item.moreInfoUrl} />
                </span>
              : null
            }
            {item.action
              ? <Link 
                  to={item.action}
                  className='ml2'
                >
                  <Icon
                    type='arrow-right'
                    className='blue'
                  />
                </Link>
              : null
            }
          </div>
        </div>

      </div>
    ))}
  </div>
)

Checklist.propTypes = {
  items: PropTypes.arrayOf(PropTypes.shape({
    isComplete: PropTypes.bool.isRequired,
    description: PropTypes.string.isRequired,
    actionUrl: PropTypes.string,
    moreInfoUrl: PropTypes.string,
  })).isRequired,
}

export default Checklist
