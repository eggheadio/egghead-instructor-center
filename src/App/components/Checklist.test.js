import React from 'react'
import renderer from 'react-test-renderer'

import examples from '../../App/utils/examples'

import Checklist from './Checklist'

test('Default', () => {
  const tree = renderer.create(<Checklist items={[
    {
      isComplete: false,
      description: examples.text.short,
    },
    {
      isComplete: true,
      description: examples.text.short,
    },
  ]} />)
  expect(tree).toMatchSnapshot()
})
