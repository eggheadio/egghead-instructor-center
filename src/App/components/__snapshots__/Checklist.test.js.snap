exports[`test Default 1`] = `
<div>
  <div
    className="flex items-center b mb3">
    <div
      className="mr2">
      <span
        className="fa fa-square-o fa-lg green" />
    </div>
    <div>
      <div>
        <span
          className="green">
          Lorem ipsum
        </span>
      </div>
    </div>
  </div>
  <div
    className="flex items-center b mb3">
    <div
      className="mr2">
      <span
        className="fa fa-check-square-o fa-lg gray" />
    </div>
    <div>
      <div>
        <span
          className="strike gray">
          Lorem ipsum
        </span>
      </div>
    </div>
  </div>
</div>
`;
